index.js

```diff
@@ -99,7 +92,61 @@ async function pay (plugin, {
   }
 }
 
+async function pull (plugin, {
+  pointer,
+  amount,
+  streamOpts = {},
+  callback,
+  callbackOpts = {}
+}) {
+  await plugin.connect()
+  const receiveMax = amount || Infinity
+
+  let response
+  try {
+    response = await query(pointer)
+  } catch (err) {
+    const endpoint = new URL(pointer.startsWith('$')
+      ? 'https://' + pointer.substring(1)
+      : pointer)
+    const postResponse = await fetch(endpoint.href, {
+      method: 'post'
+    })
+    if (postResponse.status !== 200) {
+      throw new Error('got error response from spsp server.' +
+        ' endpoint="' + endpoint.href + '"' +
+        ' status=' + postResponse.status +
+        ' message="' + (await postResponse.text()) + '"')
+    }
+    response = await query(pointer)
+  }
+
+  if (response.contentType.indexOf('application/spsp4+json') !== -1) {
+    const ilpConn = await createConnection({
+      plugin,
+      destinationAccount: response.destinationAccount,
+      sharedSecret: response.sharedSecret,
+      ...streamOpts
+    })
+
+    const stream = await ilpConn.createStream()
+
+    try {
+      await stream.receiveTotal(receiveMax, { timeout: streamOpts.timeout })
+    } catch (err) {}
+
+    if (callback) {
+      callback(stream.totalReceived, callbackOpts)
+    }
+
+    await ilpConn.end()
+  } else {
+    // PSK2 Solution?
```

Comment:
should throw an error here instead of silently failing

Reply:
done